evm/vm/base.py

```diff
@@ -77,14 +80,12 @@ def apply_transaction(self, transaction):
         """
         Apply the transaction to the vm in the current block.
         """
-        computation, block, trie_data_dict = self.state.apply_transaction(
+        computation, block, receipt = self.state.apply_transaction(
```

Comment:
Reminder: the reason why `apply_transaction` returned `trie_data_dict` is for the stateless client to generate witness, which was discussed here: https://github.com/ethereum/py-evm/pull/247#issuecomment-356180921

But since the sharding roadmap was (largely) changed, now the stateless client implementation won't happen in couple months, so I don't have a strong opinion about the change. Hope that we will find a solution to solve both stateless/non-stateless client by then. :)

Reply:
We could have the VM incrementally generate the trie data, I suppose? The problem was that it was re-generating it from scratch for every transaction applied

Reply:
Except it wouldn't make the VM class stateless, so I guess that wouldn't really buy us anything

Reply:
@hwwhww we'll continue to move towards stateless clients but for now going to prioritze full sync and then we'll figure out how to make it work with stateless clients also.